orthopairs/src/main/java/org/reactome/release/orthopairs/AlternateIdMapper.java

```diff
@@ -0,0 +1,169 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.util.*;
+
+public class AlternateIdMapper {
+    private static final Logger logger = LogManager.getLogger();
+    // Since we require some species-specific databases, there is a method for mapping the returned files. This wasn't
+    // abstractble, considering that the files all have different formats.
+    public static Map<String, Set<String>> getAltIdMappingFile(Object speciesKey, String alternateIdFilename) throws IOException {
+
+        File alternateIdFile = new File(alternateIdFilename);
+        BufferedReader br = new BufferedReader(new FileReader(alternateIdFile));
+        Map<String,Set<String>> altIdToEnsemblMap = new HashMap<>();
+        if (speciesKey.equals("mmus")) {
```

Comment:
I think Java's switch/case syntax has supported switching on strings for a while. Personally, I find the switch/case a more appropriate choice for this sort of code.

Reply:
I wasn't aware of switch/case, I'll look into it

Reply:
There's a classic refactoring pattern called 'replace conditional with polymorphism' that would be perfect here.  

https://refactoring.com/catalog/replaceConditionalWithPolymorphism.html

A family of small, simple classes may be better than a switch statement with an abstract class or interface `AlternateIDMapper` and subclasses for each species that inherit from it.  You could do the mapping assignment in one line and have the map implementations more self-contained for each species.

Reply:
Yes, you're right.  I was just to show the idea.  I've usually seen this go with a factory class that is responsible for choosing, creating, and returning the correct class based on the species being processed.

https://www.tutorialspoint.com/design_pattern/factory_pattern.htm

Reply:
Yes, that would move the logic out of the AlternateIdMapper and into a factory.

Reply:
Yup, but it would allow the methods that are specific to each species processing of alternate ids to have its own class.  It is good to keep the amount of responsibility specific and focused for any one class.